### PR TITLE
Add Azure Cognitive Search vector database 

### PR DESCRIPTION
**Pull request by farzad528**
@pablocastro please review

### CODE REVIEW

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 1]
```diff
@@ -0,0 +1,624 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "! pip install wget\n",
+    "! pip install --index-url=https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/ azure-search-documents==11.4.0a20230509004"
+   ]
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:32]: removed AzureKeyCredential

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 2]
```diff
@@ -0,0 +1,624 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 21,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "openai.api_type = \"azure\"\n",
+    "openai.api_base = \"YOUR_AZURE_OPENAI_ENDPOINT\"\n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:86]: @pablocastro all placeholders use "_" and sample hard-coded values use "-"

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 3]
```diff
@@ -0,0 +1,624 @@
...
+     "execution_count": 11,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "article_df = pd.read_csv('../../data/vector_database_wikipedia_articles_embedded.csv')  \n",
+    "  \n",
+    "# Read vectors from strings back into a list using json.loads  \n",
+    "article_df[\"title_vector\"] = article_df.title_vector.apply(json.loads)  \n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:282]: removed "literal_eval" due to security risk

Comment (farzad528) [reply examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:282]: removed intermediate json

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 4]
```diff
@@ -0,0 +1,624 @@
...
+     ]
+    }
+   ],
+   "source": [
+    "# Configure a search index\n",
+    "index_client = SearchIndexClient(\n",
+    "    endpoint=search_service_endpoint, credential=credential)\n",
+    "fields = [\n",
+    "    SimpleField(name=\"id\", type=SearchFieldDataType.String),\n",
+    "    SimpleField(name=\"vector_id\", type=SearchFieldDataType.String, key=True),\n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:315]: "id" and "vector_id" are both used in all the samples, I just included both to align with the rest

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 5]
```diff
@@ -0,0 +1,624 @@
...
+    "    SimpleField(name=\"url\", type=SearchFieldDataType.String),\n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:316]: modified "url" to SimpleField vs Searchable

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 6]
```diff
@@ -0,0 +1,624 @@
...
+    "    SearchableField(name=\"title\", type=SearchFieldDataType.String),\n",
+    "    SearchableField(name=\"text\", type=SearchFieldDataType.String),\n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:318]: removed instances of "searchable=true" for SearchableFields

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 7]
```diff
@@ -0,0 +1,624 @@
...
+    "    embeddings = response['data'][0]['embedding']\n",
+    "    return embeddings\n",
+    "\n",
+    "# Pure Vector Search\n",
+    "query = \"modern art in Europe\"  \n",
+    "  \n",
+    "search_client = SearchClient(search_service_endpoint, index_name, AzureKeyCredential(search_service_api_key))  \n",
+    "  \n",
+    "results = search_client.search(  \n",
+    "    search_text=None,  \n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:539]: Replaced empty "" with None

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 8]
```diff
@@ -0,0 +1,624 @@
...
+      "\n",
+      "Title: Scottish\n",
+      "Score: 0.016393441706895828\n",
+      "URL: https://simple.wikipedia.org/wiki/Scottish\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Hybrid Search\n",
```
Comment (farzad528) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:583]: corrected type

File: examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb [hunk 9]
```diff
@@ -2409,13 +2460,250 @@
...
+    "    SearchableField(name=\"text\", type=SearchFieldDataType.String),\n",
+    "    SearchField(name=\"title_vector\", type=SearchFieldDataType.Collection(SearchFieldDataType.Single),\n",
+    "                searchable=True, dimensions=1536, vector_search_configuration=\"my-vector-config\"),\n",
+    "    SearchField(name=\"content_vector\", type=SearchFieldDataType.Collection(SearchFieldDataType.Single),\n",
+    "                searchable=True, dimensions=1536, vector_search_configuration=\"my-vector-config\"),\n",
+    "]\n",
+    "\n",
+    "vector_search = VectorSearch(\n",
+    "    algorithm_configurations=[\n",
+    "        VectorSearchAlgorithmConfiguration(\n",
```
Comment (pablocastro) [inline examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb:2565]: Is this still right in the latest build of the SDK? In the C# version we did per-algo kind types, don't remember if we did the same in Python. Same with the hnsw_parameters property below.

**Review by pablocastro — APPROVED** [review]
LGTM other than looking at using a more recent nightly build of the Python SDK and making sure things still work end to end since there were a few last changes coming into the SDK API.

File: examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb [hunk 10]
```diff
@@ -84,6 +90,8 @@
     "!pip install redis\n",
     "!pip install typesense\n",
     "!pip install clickhouse-connect\n",
+    "# TODO: Update to latest pypi package once released\n",
+    "!pip install --index-url=https://pkgs.dev.azure.com/azure-sdk/public/_packaging/azure-sdk-for-python/pypi/simple/ azure-search-documents==11.4.0a20230509004\n",
```
[the remark this answers is not in the document]
Comment (farzad528) [reply examples/vector_databases/Using_vector_databases_for_embeddings_search.ipynb:94]: Ack, will wait till EOB 07/11 and update.

**Comment by farzad528** [comment]
@ted-at-openai @colin-openai @logankilpatrick please approve

**Comment by simonpfish** [comment]
Hey all. Thank you for the contribution. This looks good, although I also think this example could be more helpful if it showed people how to create the embeddings themselves as well. 

There are a couple of merge conflicts that need to be resolved as well before merging.

**Comment by farzad528** [comment]
> Hey all. Thank you for the contribution. This looks good, although I also think this example could be more helpful if it showed people how to create the embeddings themselves as well.
> 
> There are a couple of merge conflicts that need to be resolved as well before merging.

@simonpfish thanks for your comments! hope this works better!
1) added a brief block on how to generate embeddings when loading data into your index
2) resolved conflicts

**Review by chuwik — APPROVED** [review]

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb [hunk 11]
```diff
@@ -0,0 +1,754 @@
...
...0538108349, -0.018146580085158348, -0.004405933897942305, 0.019664108753204346, 0.022074300795793533, 0.049479082226753235, 0.02879478596150875, -0.035323984920978546, 0.02887129969894886, -0.006372982170432806, 0.029075337573885918, 0.012082843109965324, -0.00911792367696762, -0.0062103900127112865, -0.018108323216438293, -0.017661992460489273, -0.005014857742935419, 0.00018431116768624634, -0.01938355714082718, -0.03833353519439697, 0.005824631545692682, -0.02093934267759323, -0.0020180579740554094, -0.018401628360152245, 0.005990412086248398, 0.011974448338150978, -0.01449941098690033, 0.0213856752961874, 0.011636510491371155, -0.01831236109137535, -0.019944660365581512, 0.012930873781442642, -0.013951060362160206, -0.008569573052227497, -0.019791632890701294, 0.003940473310649395, -0.010131734423339367, -0.023681096732616425, -0.008193379268050194, 0.006516445893794298, -0.00263017020188272, -0.00646862480789423, -0.01623372919857502, 0.015302808955311775, 0.0245865136384964, -0.023757610470056534, -0.02056952565908432, -0.01583840698003769, -0.009277327917516232, -0.008442049846053123, 0.0007037697941996157, 0.02059503085911274, -0.019855394959449768, -0.009181685745716095]\n"
```
Comment (simonpfish) [inline examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb:530]: Printing the whole vector doesn't seem necessary, perhaps the first few items would be good enough.

**Review by simonpfish — APPROVED** [review]

File: examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb (hunk 11 above)
Comment (farzad528) [reply examples/vector_databases/azuresearch/Getting_started_with_azure_cognitive_search_and_openai.ipynb]: cleaned up!

**Comment by farzad528** [comment]
Hi @simonpfish can you merge :)

**Comment by simonpfish** [comment]
Merged! Thank you for your contribution :)

**Pull request merged**